uwuno/src/main/java/com/learning/uwuno/services/gameService.java

```diff
@@ -48,34 +49,41 @@ public player addPlayer(String name, String uid) {
         return roomList;
     }
 
-    // Callees must handle NoSuchElementException
     public room getRoom(String uid) {
-        return roomList.stream().filter(t -> t.getUid().equals(uid)).findFirst().get();
+        return roomList.stream().filter(t -> t.getUid().equals(uid)).findFirst()
+                .orElseThrow(() -> new errorNotFound("Room ID could not be found"));
     }
 
     public player getPlayer(String uid, String pid) {
-        return getRoom(uid).getPlayer(pid);
+        try {
+            return getRoom(uid).getPlayer(pid);
+        }
+        catch (NoSuchElementException e) {
+            throw new errorNotFound("Player ID could not be found");
+        }
     }
 
     // PUTS
     public void updateRoomName(String uid, String newName) {
         if (newName.isBlank())
-            throw new badRequest();
-        room room = roomList.stream().filter(t -> t.getUid().equals(uid)).findFirst().get();
+            throw new badRequest("Room name cannot be blank");
+        room room = getRoom(uid);
         room.setRoomName(newName);
     }
 
     public void updateRoomStatus(String uid, String status) {
         if (status.isBlank())
-            throw new badRequest();
-        room room = roomList.stream().filter(t -> t.getUid().equals(uid)).findFirst().get();
+            throw new badRequest("Status cannot be blank");
+        room room = getRoom(uid);
         room.Status roomStatus = utils.stringToRoomState(status);
 
         if (validStatusChange(room, roomStatus)) {
             room.setRoomStatus(roomStatus);
             setUpGameState(room, roomStatus);
-        } else {
-            throw new badRequest();
+        }
+        else {
+            // Temp
+            throw new badRequest("Failed to change room status");
```

Comment:
@achang1 I missed it originally but are we planning to use the validStatusChange() anywhere else + for other purposes.
If not I'm going to just make it throw instead the switch, as of now I don't really get how we can use those functions elsewhere?

Reply:
For now, I'm not sure because we haven't gotten to the other state changes yet. We might need to check if the previous state is valid before changing. We can change it later, after we reached that point. But up to you.